src/ng/q.js

```diff
@@ -254,25 +256,30 @@ function qFactory(nextTick, exceptionHandler) {
    *
    * @returns {Deferred} Returns a new instance of deferred.
    */
-  var defer = function() {
-    var d = new Deferred();
+  var defer = function(trackPromise) {
+    var d = new Deferred(trackPromise);
     //Necessary to support unbound execution :/
     d.resolve = simpleBind(d, d.resolve);
     d.reject = simpleBind(d, d.reject);
     d.notify = simpleBind(d, d.notify);
     return d;
-  };
+  }
 
-  function Promise() {
+  function Promise(trackPromise) {
     this.$$state = { status: 0 };
+  
+    this.trackPromise = typeof(trackPromise) !== 'undefined' ? trackPromise : true;
```

Comment:
This can just be 
trackPromise ? trackPromise: true

Reply:
or you can also do 
trackPromise !== undefined ? trackPromise : true

There is no need to do the typeof operator